lib/virt_disk/block_file.rb

```diff
@@ -1,9 +1,10 @@
 module VirtDisk
   class BlockFile
-    attr_accessor :path
+    attr_accessor :path, :offset
 
-    def initialize(path)
+    def initialize(path, offset=0)
```

Comment:
@movitto The examples you give illustrate my point. The higher-level partition code maintains the offset. This code should only deal with absolute block addresses.